.github/workflows/stack.yml

```diff
@@ -60,31 +60,38 @@ jobs:
         stack-version: ${{ matrix.stack-ver }}
         enable-stack: true
       uses: haskell/actions/setup@v1
+      id: haskell-setup
+    - run: |
+        echo "runner.os         = ${{ runner.os                               }}"
+        echo "OSTYPE            = ${{ env.OSTYPE                              }}"
+        echo "ghc-path          = ${{ steps.haskell-setup.outputs.ghc-path    }}"
+        echo "ghc-exe           = ${{ steps.haskell-setup.outputs.ghc-exe     }}"
+        echo "cabal-path        = ${{ steps.haskell-setup.outputs.cabal-path  }}"
+        echo "cabal-exe         = ${{ steps.haskell-setup.outputs.cabal-exe   }}"
+        echo "cabal-store       = ${{ steps.haskell-setup.outputs.cabal-store }}"
+        echo "stack-path        = ${{ steps.haskell-setup.outputs.stack-path  }}"
+        echo "stack-exe         = ${{ steps.haskell-setup.outputs.stack-exe   }}"
+        echo "stack-root        = ${{ steps.haskell-setup.outputs.stack-root  }}"
+        echo "STACK_ROOT (orig) = ${{ env.STACK_ROOT                          }}"
+        echo "STACK_ROOT=${{ steps.haskell-setup.outputs.stack-root }}" >> ${GITHUB_ENV}
+        echo "STACK_VER=$(stack --numeric-version)"                     >> ${GITHUB_ENV}
+      name: Information about the Haskell setup
+    - run: |
+        echo "STACK_ROOT (fix)  = ${STACK_ROOT}"
+        echo "STACK_VER         = ${STACK_VER}"
+      name: More information about the Haskell setup
     - run: |
         stack update --silent
-        if [[ "$OSTYPE" == "linux-gnu"* || "$OSTYPE" == "darwin"* ]]; then
-          echo "system-ghc: true" >> ~/.stack/config.yaml
-        else
-          echo "system-ghc: true" >> ${STACK_ROOT}/config.yaml
```

Comment:
Oops, I thought the environment variable `STACK_ROOT` was windows-specific. Good Job.

Reply:
Well, you were right.  Indeed `STACK_ROOT` is set only on Windows.  This is why I have added some steps to set it on all runners, using the `stack-root` output from the setup action.

Reply:
A-ha. I see. The previous step does not just print information but also set up this variable.